org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/handlers/SignatureHelpHandler.java

```diff
@@ -96,20 +107,34 @@ public SignatureHelp signatureHelp(TextDocumentPositionParams position, IProgres
 					int activeParameter = currentParameter < 0 ? 0 : currentParameter;
 					if (node != null) {
 						if (help2 != null) {
-							for (int i = 0; i < infos.size(); i++) {
-								if (infos.get(i).getParameters().size() >= size) {
-									CompletionProposal proposal = collector.getInfoProposals().get(infos.get(i));
-									IMethod m = JDTUtils.resolveMethod(proposal, unit.getJavaProject());
-									if (isSameParameters(m, help2, collector2, unit.getJavaProject())) {
-										help.setActiveSignature(i);
-										help.setActiveParameter(activeParameter);
-										break;
+							if (method != null) {
+								for (int i = 0; i < infos.size(); i++) {
+									if (infos.get(i).getParameters().size() >= size) {
+										CompletionProposal proposal = collector.getInfoProposals().get(infos.get(i));
+										IMethod m = JDTUtils.resolveMethod(proposal, unit.getJavaProject());

```

Comment:
you might want to store unit.getJavaProject() in a local variable, as it's not a simple reference to the java project, but the implementation cycles through the parent hierarchy every time, that's wasted CPU.